hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/Application.java

```diff
@@ -126,171 +116,157 @@ public static void main(final String[] args) {
     @Override
     public void run(final ApplicationArguments applicationArguments) throws Exception {
         final long startTime = System.currentTimeMillis();
+        ExitCodeType detectExitCode = ExitCodeType.SUCCESS;
 
         try {
-            final boolean runDetect = initializeRun(applicationArguments.getSourceArgs());
-
-            if (runDetect) {
-                final DetectProject detectProject = detectProjectManager.createDetectProject();
-                logger.info("Project Name: " + detectProject.getProjectName());
-                logger.info("Project Version Name: " + detectProject.getProjectVersion());
-                if (!detectConfigWrapper.getBooleanProperty(DetectProperty.BLACKDUCK_HUB_OFFLINE_MODE)) {
-                    final ProjectVersionView projectVersionView = hubManager.updateHubProjectVersion(detectProject);
-                    hubManager.performPostHubActions(detectProject, projectVersionView);
-                } else if (!detectConfigWrapper.getBooleanProperty(DetectProperty.DETECT_HUB_SIGNATURE_SCANNER_DISABLED)) {
-                    hubSignatureScanner.scanPathsOffline(detectProject);
-                }
-
-                for (final ExitCodeReporter exitCodeReporter : exitCodeReporters) {
-                    exitCodeType = ExitCodeType.getWinningExitCodeType(exitCodeType, exitCodeReporter.getExitCodeType());
-                }
-            }
-        } catch (final Exception e) {
-            populateExitCodeFromExceptionDetails(e);
-        } finally {
-            cleanupRun();
-        }
+            detectInfo.init();
```

Comment:
If we are using SLAP here then all of the initialization and if's should be in an separate method outside of the ru method, just like runDetect(), OR we can move the code from runDetect into the run method.